src/components/Timer.react.js

```diff
@@ -0,0 +1,322 @@
+import {includes} from 'ramda';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react'; // eslint-disable-line no-unused-vars
+
+function getFormat(format_type) {
+    const formats = {
+        default: {},
+        verbose: {verbose: true},
+        colonNotation: {colonNotation: true},
+        compact: {compact: true},
+    };
+    return formats[format_type];
+}
+
+/**
+ * The Timer component has all the functionality of the Interval component plus
+ * the following additional features:
+ *
+ * Operate in either `countdown` or `stopwatch` (count up) modes.
+ * Display custom messages, or start/stop jobs at specified times.
+ * Convert milliseconds into human readable times.
+ * Update messages clientside to help improve app performance.
+ * Specify the elapsed times to fire a callback rather than every interval
+ **/
+
+export default class Timer extends Component {
+    constructor(props) {
+        super(props);
+        this.intervalId = null;
+        this.renderMessage = null;
+        this.reportInterval = this.reportInterval.bind(this);
+        this.handleTimer = this.handleTimer.bind(this);
+        this.handleMessages = this.handleMessages.bind(this);
+
+        // this.countdown is used to handle the timer within this component.  If it's in stopwatch mode
+        //   then the count-up time is calculated just prior to setProps({timer:})
+        this.countdown = 0;
+    }
+
+    handleTimer(props) {
+        const {
+            n_intervals,
+            max_intervals,
+            disabled,
+            rerun,
+            interval,
+        } = this.props;
+
+        // Check if timer should stop or shouldn't even start
+        if (
+            max_intervals === 0 ||
+            disabled ||
+            (n_intervals >= max_intervals && max_intervals !== -1) ||
+            (this.countdown === 0 && !rerun) ||
+            (this.countdown > 0 && this.countdown < interval)
+        ) {
+            // stop existing timer
+            if (this.intervalId) {
+                this.clearTimer();
+            }
+            // and don't start a timer
+            return;
+        }
+
+        // keep the existing timer running
+        if (this.intervalId) {
+            if (this.countdown === 0 && rerun) {
+                this.initTimer(props);
+            }
+            return;
+        }
+
+        // it hasn't started yet (& it should start)
+        this.intervalId = window.setInterval(this.reportInterval, interval);
+    } // end handle timer
+
+    handleMessages(props, new_timer) {
+        const {messages, timer_format} = this.props;
+
+        const messagesObj = Object.assign({}, messages);
+        if (new_timer in messagesObj) {
+            this.renderMessage = `${messagesObj[new_timer]}`;
+        }
+
+        const prettyMilliseconds = require('pretty-ms');
```

Comment:
Put all imports at the top - and you should be able to convert this to the more modern `import prettyMilliseconds from 'pretty-ms';`

Reply:
OK, I fixed this